contracts/Account.sol

```diff
@@ -695,4 +695,22 @@ contract Account is UUPSOwnableUpgradeable, UsingRegistryUpgradeable, Managed, I
 
         return (pendingWithdrawal.value, pendingWithdrawal.timestamp);
     }
+
+    /**
+     * @notice Votes on a proposal in the referendum stage.
+     * @param proposalId The ID of the proposal to vote on.
+     * @param index The index of the proposal ID in `dequeued`.
+     * @param yesVotes The yes votes weight.
+     * @param noVotes The no votes weight.
+     * @param abstainVotes The abstain votes weight.
+     */
+    function voteProposal(
+        uint256 proposalId,
+        uint256 index,
+        uint256 yesVotes,
+        uint256 noVotes,
+        uint256 abstainVotes
+    ) public onlyManager {
+        getGovernance().votePartially(proposalId, index, yesVotes, noVotes, abstainVotes);
```

Comment:
It will fail even with current approach (It requires for proposal to be dequeued, which is not valid for proposal that doesn't exist). But for transparency I added the check